pkg/spegel/spegel.go

```diff
@@ -188,12 +187,12 @@ func (c *Config) Start(ctx context.Context, nodeConfig *config.Node) error {
 		libp2p.Identity(p2pKey),
 		libp2p.Peerstore(ps),
 		libp2p.PrivateNetwork(c.PSK),
-		libp2p.PrometheusRegisterer(metrics.DefaultRegisterer),
 	}
-	router, err := routing.NewP2PRouter(ctx, routerAddr, c.Bootstrapper, c.RegistryPort, registryScheme, opts...)
```

Comment:
I think we need to retain the passing of the scheme here, don't we? Otherwise spegel wants to use http instead of https.

Reply:
I came up with a different solution by checking if the incoming request uses TLS or not. This way it does not need to be explicitly set.

https://github.com/XenitAB/spegel/blob/d84d7710aa733280f4923d5b8ee96d624c54fd0a/pkg/registry/registry.go#L230-L233

Reply:
ah interesting. so if the client request is over https, it makes the proxy connection over https as well. that works!